File: TA/Frontend/RHStoreWS/RHStoreComprasBO/PrendaSeleccionadaBO.cs
using RHStoreBaseBO;
using RHStoreBaseBO.ServiciosWeb;
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace RHStoreComprasBO
{
    public class PrendaSeleccionadaBO : BaseBO
    {
        /*
        public int insertar(prendaSeleccionada prenda)
        {
            return this.PrendaSeleccionadaWS.insertarPrendaSeleccionada(prenda);
        }
        */

        public int insertar(int idPrenda, int fidCarrito, int cantidad, double precio)
        {
            return this.PrendaSeleccionadaWS.insertarPrendaSeleccionada(idPrenda, fidCarrito, cantidad, precio);
        }

        public int modificar(prendaSeleccionada prenda)
        {
            return this.PrendaSeleccionadaWS.modificarPrendaSeleccionada(prenda);  //la prenda ya modificada
        }
        public int eliminar(int idPrenda, int idCarrito)
        {
            return this.PrendaSeleccionadaWS.eliminarPrendaSeleccionada(idPrenda, idCarrito);
        }

        public prendaSeleccionada obtenerPorId(int idPrenda, int idCarrito)
        {
            return this.PrendaSeleccionadaWS.obtenerPrendaSeleccionadaPorId(idPrenda, idCarrito);
        }
        public BindingList<prendaSeleccionada> listarPrendasPorCarrito(int idCarrito)
        {
            prendaSeleccionada[] prendas = this.PrendaSeleccionadaWS.listarPrendasIdCarrito(idCarrito);
            return new BindingList<prendaSeleccionada>(prendas != null ? prendas.ToList() : new List<prendaSeleccionada>());
        }

        public BindingList<prendaSeleccionada> listarPrendasIdOrden(int idOrden)
        {
            prendaSeleccionada[] prendas = this.PrendaSeleccionadaWS.listarPrendasIdOrden(idOrden);
            return new BindingList<prendaSeleccionada>(prendas != null ? prendas.ToList() : new List<prendaSeleccionada>());
        }
        public int aplicarCuponLista(int idCarrito, double descuento)
        {
            return this.PrendaSeleccionadaWS.aplicarCuponLista(idCarrito, descuento);
        }

    }
}

File: TA/Backend/SoftRhRMI_Servidor/src/pe/edu/pucp/softrh/rmi/interfaces/PrendaSeleccionadaBO.java
package pe.edu.pucp.softrh.rmi.interfaces;

import java.rmi.Remote;
import java.rmi.RemoteException;
import java.util.ArrayList;
import pe.edu.pucp.softrh.compras.model.PrendaSeleccionada;

public interface PrendaSeleccionadaBO extends Remote{
    Integer insertar(Integer idPrenda, Integer fidCarrito, Integer cantidad, Double precio) throws RemoteException;

    Integer modificar(PrendaSeleccionada prenda) throws RemoteException;

    Integer eliminar(Integer idPrenda, Integer idCarrito) throws RemoteException;

    PrendaSeleccionada obtenerPorIdPrenda(Integer idPrenda, Integer idCarrito) throws RemoteException;

    ArrayList<PrendaSeleccionada> obtenerPorIdCarrito(Integer idCarrito) throws RemoteException;

    ArrayList<PrendaSeleccionada> obtenerPorIdOrden(Integer idOrden) throws RemoteException;

    Integer aplicarCuponLista(Integer idCarrito, Double descuento) throws RemoteException;
}
